BraveWallet/Crypto/BuySwapSwap/SendTokenView.swift

```diff
@@ -71,6 +80,15 @@ struct SendTokenView: View {
                                                      sendTokenStore.selectedSendToken?.symbol ?? ""),
                     text: $amountInput
           )
+            .onChange(of: amountInput, perform: { value in
+              guard let token = sendTokenStore.selectedSendToken else { return }
+              // will ignore if input is not a decimal number
+              if let decimal = Decimal(string: value), -decimal.exponent > 0 {
+                if -decimal.exponent > Int(token.decimals) {
+                  self.amountInput = String(value.prefix(Int(token.decimals) + 2))
```

Comment:
Is it possible for this to cause recursion in any way since you're changing the amount input in a closure that is triggered when you change the amount input? Not sure if `onChange` is triggered if the value is changed within the closure

Reply:
> Is it possible for this to cause recursion in any way since you're changing the amount input in a closure that is triggered when you change the amount input? Not sure if `onChange` is triggered if the value is changed within the closure

It will trigger `onChange` one more time, but it will fail at line 87